lib/broccoli/ember-app.js

```diff
@@ -1534,25 +1536,29 @@ EmberApp.prototype.toTree = function(additionalTrees) {
  */
 EmberApp.prototype.contentFor = function(config, match, type) {
   var content = [];
-  var deprecatedHooks = ['app-prefix', 'app-suffix'];
+  var privateHooks = ['app-config'];
+  var deprecatedHooks = ['app-prefix', 'app-suffix', 'config-module'];
   var deprecate = this.project.ui.writeDeprecateLine.bind(this.project.ui);
 
   switch (type) {
-    case 'head':             this._contentForHead(content, config);           break;
-    case 'config-module':    this._contentForConfigModule(content, config);   break;
-    case 'app-boot':         this._contentForAppBoot(content, config);        break;
-    case 'test-body-footer': this._contentForTestBodyFooter(content); break;
+    case 'head':             this._contentForHead(content, config);         break;
+    case 'app-config':       this._contentForAppConfig(content, config);    break;
+    case 'app-boot':         this._contentForAppBoot(content, config);      break;
+    case 'test-body-footer': this._contentForTestBodyFooter(content);       break;
   }
 
-  content = this.project.addons.reduce(function(content, addon) {
-    var addonContent = addon.contentFor ? addon.contentFor(type, config, content) : null;
-    if (addonContent) {
-      deprecate('The `' + type + '` hook used in ' + addon.name + ' is deprecated. The addon should generate a module and have consumers `require` it.', !~deprecatedHooks.indexOf(type));
-      return content.concat(addonContent);
-    }
+  // Don't leak private content hook types.
+  if (!~privateHooks.indexOf(type)) {
```

Comment:
Although this is accurate, I had to think about it when reading. We should either convert `privateHooks` to a lookup table, or use `!== -1`